### PR TITLE
docs: add description of resolution order of parserOptions.project

### PR DESCRIPTION
**Pull request by kazizi55**
## PR Checklist

- [x] Addresses an existing open issue: fixes https://github.com/typescript-eslint/typescript-eslint/issues/5617
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

- add description of resolution order of parserOptions.project

### CODE REVIEW

File: docs/linting/typed-linting/MONOREPOS.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ Be sure to update your `.eslintrc.js` to point at this new config file.
 
 The `parserOptions.project` option introduced in [Linting with Type Information](../TYPED_LINTING.md) accepts an array of relative paths.
 Paths may be provided as [Node globs](https://github.com/isaacs/node-glob/blob/f5a57d3d6e19b324522a3fa5bdd5075fd1aa79d1/README.md#glob-primer).
+And the path resolution is performed in order from the beginning of the array.
```
Comment (JoshuaKGoldberg) [inline docs/linting/typed-linting/MONOREPOS.md:43]: Hmm, this isn't quite saying what I think we want it to. What happens is that the _first matching project_ is what gets used.

1. Normalizing project input paths: https://github.com/typescript-eslint/typescript-eslint/blob/09c04fb0d5a5dee0dc266abb2e21aa9a1c489712/packages/typescript-estree/src/parser.ts#L308-L311
2. Non-single-run usage: https://github.com/typescript-eslint/typescript-eslint/blob/09c04fb0d5a5dee0dc266abb2e21aa9a1c489712/packages/typescript-estree/src/create-program/useProvidedPrograms.ts#L22-L28

How does this phrasing look to you?

```suggestion
For each file being linted, the first matching project path will be used as its backing TSConfig.
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
A good start! 🚀

File: docs/linting/typed-linting/MONOREPOS.md (hunk 1 above)
Comment (kazizi55) [reply docs/linting/typed-linting/MONOREPOS.md:43]: @JoshuaKGoldberg 
Thank you for explaining in detail!
Now I understand that the phrasing you suggested is better!

**Review by JoshuaKGoldberg — APPROVED** [review]
Super, thanks for the collaboration @kazizi55! 🙌

**Pull request merged**